### PR TITLE
feat(FOROME-792): add all, not mods for enum modals

### PR DESCRIPTION
**Pull request by Lapk1n**
PR into fix/FOROME-634

https://quantori.atlassian.net/browse/FOROME-792

### CODE REVIEW

File: src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.store.ts [hunk 1]
```diff
@@ -95,14 +123,35 @@ class ModalFiltersStore {
...
+
+    if (currentGroup) {
+      const subGroups = Object.values(dtreeStore.getQueryBuilder)
+      const { groupName } = modalEditStore
+
+      const selectedGroup: StatList | undefined = subGroups
+        .find(subGroup => subGroup.find(element => element.name === groupName))
+        ?.find(element => element.name === groupName)
+
+      this.currentGroupSubKind = selectedGroup?.['sub-kind'] as SubKinds
```
Comment (QSdmitrioul) [inline src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.store.ts]: seems that you can have exception on trying cast undefined(from safe calling) to `SubKinds`

File: src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.tsx [hunk 2]
```diff
@@ -89,7 +101,15 @@ export const ModalFilters = observer((): ReactElement => {
           {dtreeStore.selectedFilters.length || 0} {t('dtree.selected')}
         </div>
 
-        <FiltersMods />
+        <AllNotMods
+          isAllModeChecked={modalFiltersStore.currentMode === ModeTypes.All}
+          isNotModeChecked={modalFiltersStore.currentMode === ModeTypes.Not}
+          isAllModeDisabled={dtreeStore.selectedFilters.length < 2}
+          isNotModeDisabled={dtreeStore.selectedFilters.length === 0}
```
Comment (QSdmitrioul) [inline src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.tsx]: !dtreeStore.selectedFilters.length

File: src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.store.ts (hunk 1 above)
Comment (Lapk1n) [reply src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.store.ts]: thanks

**Review by ghost — APPROVED** [review]

File: src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.store.ts [hunk 3]
```diff
@@ -95,14 +123,35 @@ class ModalFiltersStore {
...
 
-    this.originGroupList = toJS(dtreeStore.selectedGroups[2]) ?? []
+    this.resetCurrentMode()
+    this.currentGroupSubKind = undefined
   }
 
-  get filteredGroupList(): [string, number][] {
-    const originGroupList: [string, number][] =
-      toJS(dtreeStore.selectedGroups[2]) ?? []
+  get originGroupList(): [string, number][] {
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.store.ts]: access modifiers

File: src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.tsx [hunk 4]
```diff
@@ -51,6 +50,19 @@ export const ModalFilters = observer((): ReactElement => {
...
+      () => dtreeStore.selectedFilters,
+      () => {
+        if (dtreeStore.selectedFilters.length < 2) {
+          modalFiltersStore.resetCurrentMode()
+        }
+      },
+    )
+
+    return () => dispose()
+  }, [currentGroup])
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.tsx]: you subscribe the reaction on selectedFilters, so it has nothing in common with currentGroup. Reaction subscribes only once - on the mount stage and then you are good to go. So useEffect deps must be empty in this case.

Comment (Lapk1n) [reply src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.tsx]: Probably overlooked that, thanks

**Review by AndreyUstyumenko — APPROVED** [review]

File: src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.store.ts [hunk 5]
```diff
@@ -95,30 +123,55 @@ class ModalFiltersStore {
...
+
+    const currentGroup =
+      dtreeStore.stepData[currentStepIndex].groups[currentGroupIndex]
+
+    if (currentGroup) {
+      const subGroups = Object.values(dtreeStore.getQueryBuilder)
+      const { groupName } = modalEditStore
 
-    return originGroupList.filter((variant: [string, number]) =>
+      const selectedGroup: StatList | undefined = subGroups
```
Comment (AlMaQntr) [inline src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.store.ts]: `modealEditStore.attributeStatustoChange`?

File: src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.tsx [hunk 6]
```diff
@@ -51,6 +50,24 @@ export const ModalFilters = observer((): ReactElement => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [currentGroupIndex])
 
+  useEffect(() => {
+    const dispose = reaction(
```
Comment (AlMaQntr) [inline src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.tsx:54]: Can we put this reaction to store?

**Review by AlMaQntr — APPROVED** [review]

File: src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.tsx (hunk 6 above)
Comment (Lapk1n) [reply src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.tsx:54]: Will be put in another PR

**Pull request merged**